R/02-filter.R

```diff
@@ -40,19 +40,27 @@ gimap_filter <- function(.data = NULL,
 #' Create a filter for pgRNAs which have a raw count of 0 for any sample/time point
 #' @description This function flags and reports which and how many pgRNAs have a raw count of 0 for any sample/time point
 #' @param gimap_dataset The special gimap_dataset from the `setup_data` function which contains the raw count data
+#' @param filter_zerocount_target_col default is NULL; Which sample column(s) should be used to check for counts of 0? If NULL and not specified, downstream analysis will select all sample columns
 #' @importFrom magrittr %>%
 #' @importFrom dplyr mutate
 #' @importFrom purrr reduce map
 #' @return a named list with the filter `filter` specifying which pgRNA have a count zero for at least one sample/time point and a report df `reportdf` for the number and percent of pgRNA which have a count zero for at least one sample/time point
 #' @examples \dontrun{
 #'   gimap_dataset <- get_example_data("gimap")
 #'   qc_filter_zerocounts(gimap_dataset)
+#'   
+#'   #or to specify a different column (or set of columns to select)
+#'   qc_filter_zerocount(gimap_dataset, filter_zerocount_target_col = c(1,2))
 #' }
 #'
 
-qc_filter_zerocounts <- function(gimap_dataset){
+qc_filter_zerocounts <- function(gimap_dataset, filter_zerocount_target_col = NULL){
 
-  counts_filter <- data.frame(gimap_dataset$raw_counts) %>% map(~.x %in% c(0)) %>% reduce(`|`)
+  if (is.null(filter_zerocount_target_col)) {filter_zerocount_target_col <- c(1:ncol(gimap_dataset$raw_counts))}
+
+  #@Howard please help: should set up a check that if filter_zerocount_target_col is not null, it's an R vector (? -- the c() thing) of integers greater than or equal to 1 and less than or equal to number of possible columns in data
```

Comment:
```suggestion
if (!all(filter_zerocount_target_col %in% 1:ncol(gimap_dataset$raw_counts))) {
    stop("The columns selected do not exist. `filter_zerocount_target_col` needs to correspond to the index of the columns in `gimap_dataset$raw_counts` that you need to filter by") 
   }
```
Something like this should work

Reply:
Very cool! I love how succinct it is! Thank you